src/main/java/wolox/training/models/User.java

```diff
@@ -112,4 +120,19 @@ public void addBook(Book book) {
     public void deleteBook(Book book) {
         books.remove(book);
     }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        Preconditions.checkNotNull(password, String.format(ErrorConstants.NOT_NULL,"password"));
+        Preconditions.checkArgument(password.length() >= 6, String.format(ErrorConstants.NOT_GREATER_THAN, "password", "0"));
+
+        this.password = new BCryptPasswordEncoder().encode(password);
```

Comment:
You don't need to instantiate the BCryptPasswordEncoder, since you defined a Bean in the configuration, you should be able to Autowire it :)